dlc-oracle/src/main/scala/org/bitcoins/dlc/oracle/DLCOracle.scala

```diff
@@ -91,85 +91,207 @@ case class DLCOracle(private val extPrivateKey: ExtPrivateKeyHardened)(implicit
+                           account = rValAccount,
+                           chainType = rValueChainIndex,
+                           keyIndex = index)
+          }
+          .toVector
+
+      descriptorTLV = uninitializedDescriptor.addNonces(rValueDbs.map(_.nonce))
+
+      epoch = UInt32(maturationTime.getEpochSecond)
+
+      eventTLV = OracleEventV0TLV(publicKey, epoch, descriptorTLV, eventName)
+
+      announcementBytes = signingVersion.calcAnnouncementBytes(eventTLV)
+      announcementSignature = signingKey.schnorrSign(announcementBytes)
+
+      eventOutcomeDbs = descriptorTLV match {
+        case enum: EnumEventDescriptorV0TLV =>
+          require(rValueDbs.size == 1)
+          val nonce = rValueDbs.head.nonce
+          enum.outcomes.map { outcome =>
+            val hash = signingVersion.calcOutcomeHash(outcome)
+            EventOutcomeDb(nonce, outcome, hash)
+          }
+        case range: RangeEventDescriptorV0TLV =>
+          require(rValueDbs.size == 1)
+          val nonce = rValueDbs.head.nonce
+          range.outcomes.map { outcome =>
+            val hash = signingVersion.calcOutcomeHash(outcome.bytes)
+            EventOutcomeDb(nonce, outcome.toInt.toString, hash)
+          }
+        case largeRange: LargeRangeEventDescriptorV0TLV =>
+          val signDbs = largeRange.signedNonceOpt match {
+            case Some(signedNonce) =>
+              val plusHash = signingVersion.calcOutcomeHash("+")
+              val minusHash = signingVersion.calcOutcomeHash("-")
+              Vector(EventOutcomeDb(signedNonce, "+", plusHash),
+                     EventOutcomeDb(signedNonce, "-", minusHash))
+            case None =>
+              Vector.empty
+          }
+          val digitDbs = largeRange.digitNonces.flatMap { nonce =>
+            0.until(largeRange.base.toInt).map { num =>
+              val hash = signingVersion.calcOutcomeHash(Int32(num).bytes)
+              EventOutcomeDb(nonce, num.toString, hash)
+            }
+          }
+
+          digitDbs ++ signDbs
+        case external: ExternalEventDescriptorV0TLV =>
+          throw new IllegalArgumentException(
+            s"Unable to parse ExternalEventDescriptorV0TLVs got $external")
+      }
+
+      eventDbs = rValueDbs.map { db =>
+        EventDb(db.nonce,
+                publicKey,
+                eventName,
+                eventOutcomeDbs.size,
+                signingVersion,
+                maturationTime,
+                None,
+                announcementSignature,
+                descriptorTLV)
       }
 
-      _ <- rValueDAO.create(rValueDb)
-      eventDb <- eventDAO.create(eventDb)
+      _ <- rValueDAO.createAll(rValueDbs)
+      _ <- eventDAO.createAll(eventDbs)
       _ <- eventOutcomeDAO.createAll(eventOutcomeDbs)
-    } yield eventDb
+    } yield {
+      OracleEvent(eventDbs).announcementTLV
+    }
   }
 
-  def signEvent(nonce: SchnorrNonce, outcome: String): Future[EventDb] = {
+  def signEvent[T <: DLCAttestationType](
```

Comment:
So i think this should be renamed to something like `signSingleNonceEvent()`. 

I propse creating a new marker trait 

```scala
trait SingleNonceEvent { self: EventDescriptorTLV =>
  def nonce: SchnorrNonce
}
```

Then the  parameter should be changed to be `descriptor: SingleNonceEvent`

Reply:
Oh and then the type for 

```scala 
case class EnumEventDescriptorV0TLV(
    nonce: SchnorrNonce,
    outcomes: Vector[String])
    extends EventDescriptorTLV 
    with SingleNonceEvent
    
    
case class RangeEventDescriptorV0TLV(
    nonce: SchnorrNonce,
    start: Int32,
    count: UInt32,
    step: UInt16)
    extends EventDescriptorTLV
    with SingleNonceEvent 
```

Reply:
Idk if I like this because this is used for signing multi-nonce events as well. When doing multi-nonce events this function is just called multiple times